modules/loader/src/main/java/com/opengamma/strata/loader/csv/CdsTradeCsvPlugin.java

```diff
@@ -372,122 +337,66 @@ private static PeriodicSchedule parseSchedule(CsvRow row, Currency currency) {
+    return CdsTrade.class.getSimpleName();
+  }
 
-    @Override
-    public List<String> headers(List<CdsIndexTrade> trades) {
-      // determine what elements of trades are present
-      boolean premium = false;
-      boolean stepInOffset = false;
-      boolean settleOffset = false;
-      boolean startConv = false;
-      boolean endConv = false;
-      boolean overrideStart = false;
-      for (CdsIndexTrade trade : trades) {
-        CdsIndex cds = trade.getProduct();
-        PeriodicSchedule schedule = cds.getPaymentSchedule();
-        premium |= trade.getUpfrontFee().isPresent();
-        stepInOffset |= !cds.getStepinDateOffset().equals(DaysAdjustment.ofCalendarDays(1));
-        settleOffset |= !cds.getSettlementDateOffset().equals(
-            DaysAdjustment.ofBusinessDays(3, schedule.getBusinessDayAdjustment().getCalendar()));
-        startConv |= schedule.getStartDateBusinessDayAdjustment().isPresent();
-        endConv |= schedule.getEndDateBusinessDayAdjustment().isPresent();
-        overrideStart |= schedule.getOverrideStartDate().isPresent();
-      }
-      return createHeaders(true, premium, stepInOffset, settleOffset, startConv, endConv, overrideStart);
-    }
+  @Override
+  public Set<String> supportedTradeTypes() {
+    return ImmutableSet.of(CdsTrade.class.getSimpleName());
+  }
 
-    @Override
-    public void writeCsv(CsvRowOutputWithHeaders csv, CdsIndexTrade trade) {
-      CdsIndex product = trade.getProduct();
-      csv.writeCell(TRADE_TYPE_FIELD, "CdsIndex");
-      csv.writeCell(CDS_INDEX_ID_SCHEME_FIELD, product.getCdsIndexId().getScheme());
-      csv.writeCell(CDS_INDEX_ID_FIELD, product.getCdsIndexId().getValue());
-      String scheme = "";
-      if (product.getLegalEntityIds().stream().map(StandardId::getScheme).distinct().count() == 1) {
-        scheme = product.getLegalEntityIds().get(0).getScheme();
-      } else {
-        scheme = product.getLegalEntityIds().stream()
-            .map(StandardId::getScheme)
-            .collect(joining(";"));
-      }
-      csv.writeCell(LEGAL_ENTITY_ID_SCHEME_FIELD, scheme);
-      csv.writeCell(LEGAL_ENTITY_ID_FIELD, product.getLegalEntityIds().stream()
-          .map(StandardId::getValue)
-          .collect(joining(";")));
-      trade.getUpfrontFee().ifPresent(premium -> writeCdsPremium(csv, premium));
-      writeCdsDetails(
-          csv,
-          product.getBuySell(),
-          product.getCurrency(),
-          product.getNotional(),
-          product.getFixedRate(),
-          product.getDayCount(),
-          product.getPaymentOnDefault(),
-          product.getProtectionStart(),
-          product.getStepinDateOffset(),
-          product.getSettlementDateOffset(),
-          product.getPaymentSchedule());
-    }
+  // Restricted constructor.
+  private CdsTradeCsvPlugin() {
   }
 
   // creates the correct set of headers
-  private static List<String> createHeaders(
+  protected static Set<String> createHeaders(
```

Comment:
This is because the method is also used in CdsIndexTradeCsvPlugin.

Initially, the class `CdsTradeCsvPlugin` was containing the inner classes : `CdsTradeCsvWriter` and `CdsIndexTradeCsvWriter`. Both writers were using common methods, written in the plugin.

As I extracted the Writers to `CdsTradeCsvPlugin` and `CdsIndexTradeCsvPlugin`, I switched the methods to `protected` to be able to use them in `CdsIndexTradeCsvPlugin`.

I could either extract common methods to another class or change to package-private visibility. 

What do you think ?

Reply:
Always try to use package scope rather than `protected`